pingcap/tidb-lightning
# lightning: start the HTTP server when receiving SIGUSR1

### What problem does this PR solve? 

Allow user to dynamically start the HTTP server even if it was disabled, to allow emergency use of PProf.

### What is changed and how it works?

Added a SIGUSR1 listener, which will start the server listening from a random port when signaled (like pingcap/br#372).

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
    * `kill -USR1 pid` and check.

Side effects

Related changes

 - Need to update the documentation
